nevergrad/optimization/multiobjective/core.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import random
+import numpy as np
+import nevergrad as ng
+import nevergrad.common.typing as tp
+from .hypervolume import HypervolumeIndicator
+
+
+# pylint: disable=too-many-instance-attributes
+class HypervolumePareto:
+    """Given several functions, and threshold on their values (above which solutions are pointless),
+    this object can be used as a single-objective function, the minimization of which
+    yields a solution to the original multiobjective problem.
+
+    Parameters
+    -----------
+    upper_bounds: Tuple of float or np.ndarray
+        upper_bounds[i] is a threshold above which x is pointless if function(x)[i] > upper_bounds[i].
+
+    Notes
+    -----
+    - This function is not stationary!
+    - The minimum value obtained for this objective function is -h,
+      where h is the hypervolume of the Pareto front obtained, given upper_bounds as a reference point.
+    - The callable keeps track of the pareto_front (see attribute paretor_front) and is therefor stateful.
+      For this reason it cannot be distributed. A user can however call the multiobjective_function
+      remotely, and aggregate locally. This is what happens in the "minimize" method of optimizers.
+    """
+
+    def __init__(self, upper_bounds: tp.Optional[tp.ArrayLike] = None, auto_bound: int = 15) -> None:
+        self._auto_bound = 0
+        self._upper_bounds = np.array([-float('inf')]) if upper_bounds is None else np.array(upper_bounds, copy=False)
+        if upper_bounds is None:
+            self._auto_bound = auto_bound
+        self._pareto: tp.List[ng.p.Parameter] = []
+        self._best_volume = -float("Inf")
+        self._hypervolume: tp.Optional[HypervolumeIndicator] = None
+
+    @property
+    def num_objectives(self) -> int:
+        return self._upper_bounds.size
+
+    def add(self, parameter: ng.p.Parameter) -> float:
+        """Given parameters and the multiobjective loss, this computes the hypervolume
+        and update the state of the function with new points if it belongs to the pareto front
+        """
+        if not isinstance(parameter, ng.p.Parameter):
+            raise TypeError(f"{self.__class__.__name__}.add should receive a ng.p.Parameter, but got: {parameter}.")
+        losses = parameter.loss
+        if not isinstance(losses, np.ndarray):
+            raise TypeError(f"Parameter should have multivalue as loss, but parameter.loss={losses} ({type(losses)}).")
+        if self._auto_bound > 0:
+            self._auto_bound -= 1
+            if (self._upper_bounds > -float("inf")).all() and (losses > self._upper_bounds).all():
+                return float('inf')  # Avoid uniformly worst points
```

Comment:
@teytaud I've added this case to auto-bound:
if one point is uniformly bad compared to previous ones, do not include it
that makes the example work way better. Is it a problem though?